src/web/aws-cloudwatch/CloudWatch.jsx

```diff
@@ -1,169 +1,101 @@
-import React, { Component } from 'react';
-import PropTypes from 'prop-types';
-import { Col, Row } from 'react-bootstrap';
+import React, { useContext } from 'react';
 
 import Wizard from 'components/common/Wizard';
-import PageHeader from 'components/common/PageHeader';
 
 import StepAuthorize from './StepAuthorize';
 import StepKinesis from './StepKinesis';
 import StepHealthCheck from './StepHealthCheck';
 import StepReview from './StepReview';
-import DEFAULT_VALUES from './default_values';
 
-export default class AWSCloudWatch extends Component {
-  static propTypes = {
-    params: PropTypes.shape({
-      step: PropTypes.string,
-    }).isRequired,
-  }
-
-  logOutput = { // Demo Data until API is wired
-    full_message: '2 123456789010 eni-abc123de 172.31.16.139 172.31.16.21 20641 22 6 20 4249 1418530010 1418530070 ACCEPT OK',
-    version: 2,
-    'account-id': 123456789010,
-    'interface-id': 'eni-abc123de',
-    src_addr: '172.31.16.139',
-    dst_addr: '172.31.16.21',
-    src_port: 20641,
-    dst_port: 22,
-    protocol: 6,
-    packets: 20,
-    bytes: 4249,
-    start: 1418530010,
-    end: 1418530070,
-    action: 'ACCEPT',
-    'log-status': 'OK',
-  }
-
-  constructor(props) {
-    super(props);
-
-    this.state = {
-      visibleAdvancedOptions: false,
-      currentStep: 'authorize',
-      enabledSteps: ['authorize'],
-      formData: {
-        ...DEFAULT_VALUES,
-      },
-      logOutput: JSON.stringify(this.logOutput, null, 2),
-    };
-  }
-
-  isDisabledStep = (step) => {
-    const { enabledSteps } = this.state;
-
-    if (!enabledSteps || enabledSteps.length === 0) {
-      return true;
-    }
+import { StepsContext } from './context/Steps';
+import { FormDataContext } from './context/FormData';
+import { LogOutputContext } from './context/LogOutput';
+import TEMPORARY_LOG from './context/temporary_log';
+
+const CloudWatch = () => {
+  const {
+    availableSteps,
+    currentStep,
+    isDisabledStep,
+    setAvailableStep,
+    setCurrentStep,
+    setEnabledStep,
+  } = useContext(StepsContext);
+  const { setFormData } = useContext(FormDataContext);
+  const { setLogOutput } = useContext(LogOutputContext);
+
+  const handleStepChange = (nextStep) => {
+    setCurrentStep(nextStep);
+  };
 
-    return !enabledSteps.includes(step);
+  const handleEditClick = nextStep => () => {
+    setCurrentStep(nextStep);
   };
 
-  handleFieldUpdate = ({ target }) => {
-    const { formData } = this.state;
+  const handleFieldUpdate = ({ target }) => {
+    const id = target.name || target.id;
     const value = Object.keys(target).includes('checked') ? target.checked : target.value;
```

Comment:
You can also use `FormUtils.getValueFromInput` to retrieve an input's value based on its type.